### DISCUSSION TITLE
How to disable buffering/caching for WebDAV?

### DISCUSSION THREAD

**Discussion by ypopovych** [photoprism/photoprism#4162] [Q&A]
Hello,

I'm trying to upload files to the PhotoPrism with WebDAV and found a strange buffering behaviour.

I uploading a 1.5Gb file from my phone and PhotoPrism quickly accepts all data. Seems as it buffers data somewhere (RAM?).
When I check FS I can see that PhotoPrism is writing data to the file in import folder on the HDD. HDD is loaded by a lot of processes so write speed is much slower than network speed.
Upload client decides that upload is stuck (because it uploaded all data and then got no response) and cancels upload before PhotoPrism writes all data to the file.

How can I decrease buffer size/disable buffering, so upload can be slower, but will finish properly?

PhotoPrism works behind NGINX proxy (request buffering is disabled). I checked it without proxy and it has the same problem.

**Comment by lastzero** [comment]
Please let us know which version(s) you are using as we have been implementing changes over the last year. It would also be great if you could test the behavior when uploading directly to PhotoPrism (without nginx proxy) to see if that makes a difference. Thank you very much! 🙏

**Comment by ypopovych** [comment]
I tested direct upload. It works the same, seems as problem is not in the NGINX.

PhotoPrism version is: `231128-f48ff16ef`. It runs as podman container

**Reply by lastzero** [reply]
Thank you for testing it again! If it's indeed a buffering issue (which is well possible since I never tried uploading such large files to a resource constrained device), this should be relatively easy to fix / optimize by anyone who knows a bit of Go. I'd also be happy to take a look at it, but there are many other issues already waiting (for much longer), so I'm not sure when I'll actually get to it...

**Reply by lastzero** [reply]
One more thing you could check is whether chunked uploading is possible (may depend on the client you are using and the webdav library we use) and if so, whether this makes a difference with regard to buffering.

**Reply by ypopovych** [reply]
I tried with https://play.google.com/store/apps/details?id=com.ghisler.android.TotalCommander and it works fine. I think it uses chunked upload. I will also contact developer of the uploader app, maybe he can tweak something too.

**Reply by lastzero** [reply]
Which uploader app did you use before?

**Reply by ypopovych** [reply]
ZPush: https://play.google.com/store/apps/details?id=de.maza.zpush
